report/render_report.R

```diff
@@ -0,0 +1,49 @@
+library(tidyverse)
+library(dotenv)
+library(REDCapR)
+library(lubridate)
+library(rmarkdown)
+library(sendmailR)
+library(redcapcustodian)
+
+init_etl("render_report")
+
+if (!dir.exists("output")){
+  dir.create("output")
+}
+
+if (!interactive()) {
+  args <- commandArgs(trailingOnly = T)
+  script_name <- word(args, 2, sep = "=")
+} else {
+  script_name <- "sample_report.Rmd"
+}
+
+report_name <- word(script_name, 1, sep = "\\.")
+
+script_run_time <- set_script_run_time()
+
+output_file <- here::here(
+  "output",
+  paste0(report_name,
+  "_",
+  format(script_run_time, "%Y%m%d%H%M%S"))
+)
+
+full_path_to_output_file <- render(
+  here::here("report", script_name),
+  output_file = output_file
+)
+
+output_file_extension <- word(full_path_to_output_file, 2 , sep = "\\.")
+attachment_object <- mime_part(full_path_to_output_file, paste0(basename(output_file), ".", output_file_extension))
```

Comment:
```suggestion
attachment_object <- mime_part(full_path_to_output_file, basename(full_path_to_output_file))
```
One last nitpick, the file attached to the email has no extension.  
Not sure why your `paste0` doesn't work, but `basename` keeps the file extension anyway.

Reply:
Note, this may just be how Apple's Mail client works, but the PDF is rendered in the email with my proposed method. I don't think this would cause any problems, but I don't recall it happening with other emailed reports.

<img width="873" alt="image" src="https://user-images.githubusercontent.com/20332546/212929615-5abe5adc-f42b-4c76-8885-3ca71933f767.png">